File: source_code/functional_optional_calc.cpp
static std::optional<int>
perform_calculation(int input) {
    if (input % 2 != 0) {
        return std::nullopt;
    }
    return input * 3;
}

static void
use_optional(int input) {
    auto res = perform_calculation(input);
    if (res) {
        std::cout << *res;
    } else {
        std::cout << "error";
    }
}

File: source_code/functional_optional_calc.rs
fn perform_calculation
        (input: i32)-> Option<i32> {
    if input % 2 != 0 {
        None
    } else {
        Some(input * 3)
    }
}

fn use_optional(input: i32) {
    match perform_calculation(input) {
        Some(res) => println!("{}", res),
        None => println!("error"),
    }
}